doc/contributing/adding-v8-fast-api.md

```diff
@@ -1,91 +1,396 @@
+additional parameter should be a pointer to the `v8::CFunction` object
+constructed by the call to `CFunction::Make`.
+
+In rare cases, there may be more than one fast callback, _eg._ if the function
+accepts optional arguments. In this case, the additional parameter should be a
+reference to an array of `v8::CFunction` objects, which is used to initialize a
+`v8::MemorySpan<v8::CFunction>`:
+
+```cpp
+uint32_t FastFuncWithoutArg() {
+  return -1;
+}
+uint32_t FastFuncWithArg(const v8::FastOneByteString& s) {
+  return s.length;
+}
+static CFunction fast_func_callbacks[] = {CFunction::Make(FastFuncWithoutArg),
+                                          CFunction::Make(FastFuncWithArg)};
+
+void Initialize(Local<Object> target,
+                Local<Value> unused,
+                Local<Context> context,
+                void* priv) {
+  Environment* env = Environment::GetCurrent(context);
+  SetFastMethodNoSideEffect(context,
+                            target,
+                            "func",
+                            SlowFunc,
+                            fast_func_callbacks);
+}
+```
+
+In addition, all method bindings should be registered with the external
+reference registry. This is done by passing both the conventional callback
+pointer and the `v8::CFunction` handle to `registry->Register`.
+
+```cpp
+void RegisterExternalReferences(ExternalReferenceRegistry* registry) {
+  registry->Register(SlowIsEven);
+  registry->Register(fast_is_even);
+}
+```
+
+Omitting this step can lead to fatal exceptions if the callback ends up in a
+snapshot (either the built-in snapshot, or a user-land one). Refer to the
+[binding functions documentation](../../src/README.md#registering-binding-functions-used-in-bootstrap)
+for more information.
+
+## Type checking
+
+Non-primitive arguments (such as TypedArrays) are passed to Fast API callbacks
+as `v8::Local<v8::Value>`. However, registering a fast callback with this
+argument type signals to the V8 engine that it can invoke the fast callback
+with _any value_ as that argument.
+
+If using arguments of type `v8::Local<v8::Value>`, then it is the
+implementation's responsibility to ensure that the arguments are validated
+before casting or otherwise consuming them. This can either take place within
+the C++ callbacks themselves, or within a JavaScript wrapper function that
+performs any necessary validation before calling the bound function.
+
+## Stack-allocated objects and garbage collection
+
+The Fast API now allows access to the isolate, and allows allocation of
+`v8::Local` handles on the stack.
+
+A fast callback intending to make use of this functionality should accept a
+final argument of type `const v8::FastApiCallbackOptions&`. V8 will pass the
+isolate pointer in `options.isolate`.
+
+If a fast callback creates any `v8::Local` handles within the fast callback,
+then it must first initialize a new `v8::HandleScope` to ensure that the
+handles are correctly scoped and garbage-collected.
```

Comment:
This doesn't have any equivalent in the previous guidance, since stack access is a new feature. This seems like the correct advice – I'm fairly sure that CI has flagged exceptions from V8 relating to this specifically – but others will be more informed than me!